destination/destination_integration_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"context"
+	"testing"
+
+	"github.com/conduitio-labs/conduit-connector-nats-pubsub/config"
+	"github.com/conduitio-labs/conduit-connector-nats-pubsub/test"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/matryer/is"
+)
+
+func TestDestination_OpenSuccess(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	destination := NewDestination()
+
+	err := destination.Configure(context.Background(), map[string]string{
+		config.KeyURLs:    test.TestURL,
+		config.KeySubject: "foo_destination",
+	})
+	is.NoErr(err)
+
+	err = destination.Open(context.Background())
+	is.NoErr(err)
+
+	err = destination.Teardown(context.Background())
+	is.NoErr(err)
+}
+
+func TestDestination_OpenFail(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	destination := NewDestination()
+
+	err := destination.Configure(context.Background(), map[string]string{
+		config.KeyURLs:    "nats://localhost:6666",
+		config.KeySubject: "foo_destination",
+	})
+	is.NoErr(err)
+
+	err = destination.Open(context.Background())
+	is.True(err != nil)
+
+	err = destination.Teardown(context.Background())
+	is.NoErr(err)
+}
+
+func TestDestination_WriteOneMessage(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	destination := NewDestination()
+
+	err := destination.Configure(context.Background(), map[string]string{
+		config.KeyURLs:    test.TestURL,
+		config.KeySubject: "foo_destination_write_pubsub",
+	})
+	is.NoErr(err)
+
+	err = destination.Open(context.Background())
+	is.NoErr(err)
+
+	err = destination.Write(context.Background(), sdk.Record{
+		Payload: sdk.RawData([]byte("hello")),
+	})
+	is.NoErr(err)
+
+	err = destination.Teardown(context.Background())
+	is.NoErr(err)
+}
+
+func TestDestination_WriteManMessages(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	destination := NewDestination()
+
+	err := destination.Configure(context.Background(), map[string]string{
+		config.KeyURLs:    test.TestURL,
+		config.KeySubject: "foo_destination_write_pubsub",
+	})
+	is.NoErr(err)
+
+	err = destination.Open(context.Background())
+	is.NoErr(err)
+
+	for i := 0; i < 1000; i++ {
+		err = destination.Write(context.Background(), sdk.Record{
+			Payload: sdk.RawData([]byte("hello")),
+		})
+		is.NoErr(err)
+	}
```

Comment:
How do we check that this data has actually been written to NATS?

Reply:
I've added the check